engine/src/main/java/org/camunda/bpm/engine/impl/telemetry/TelemetryRegistry.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.telemetry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.camunda.bpm.engine.impl.telemetry.dto.ApplicationServer;
+
+public class TelemetryRegistry {
+
+  protected Map<String, Object> entries = new HashMap<>();
+  protected ApplicationServer applicationServer;
+
+  public synchronized ApplicationServer getApplicationServer() {
+    return applicationServer;
+  }
+
+  public synchronized void setApplicationServer(ApplicationServer applicationServer) {
+    this.applicationServer = applicationServer;
+  }
+
+  public void setApplicationServer(String applicationServerVersion) {
```

Comment:
This method should be also thread safe
```suggestion
  public synchronized void setApplicationServer(String applicationServerVersion) {
```